format.go

```diff
@@ -43,4 +51,23 @@ func quoteString(s string) string {
 	return strconv.Quote(s)
 }
 
+// checkNotGuarded prevents panics caused by the given value being a nil
+// pointer with a method (invoked by the given function) not guarding against
+// nil. This is usually a stringer or a error type.
+func checkNotGuarded(v interface{}, f func() string) (s string, isNil bool) {
```

Comment:
I wonder if this might be slightly better as:

    // checkStringCall calls f and returns its result, and reports
    // if the call succeeded without panicking due to a nil pointer.
    // If f panics and v is a nil pointer, it returns false.
    func checkStringCall(v interface{}, f func() string) (s string, ok bool)

then it could be used like this:

		s, ok := checkStringCall(v, v.String)
		if !ok {
			return "s<nil>"
		}
		return "s" + quoteString(s)

This does mean you'll need to duplicate the `<nil>` string, but I think the checkStringCall contract is clearer like that.

Reply:
Yes it reads better, done.